src/js/client/editprefset-prefSettingAdjuster-widget.js

```diff
@@ -0,0 +1,269 @@
+                            inputPath: "",
+                            schema: "{that}"
+                        }
+                    }
+                },
+                modelToDom: {
+                    "": {
+                        transform: {
+                            type: "gpii.devpmt.prefSettingAdjuster.valueInputToBinder",
+                            inputPath: "",
+                            schema: "{that}"
+                        }
+                    }
+                }
+            }
+        }
+    },
+    templates: {
+        initial: "editprefset-prefSettingAdjuster-widget"
+    },
+    invokers: {
+        saveUpdateValue: {
+            funcName: "gpii.devpmt.prefSettingAdjuster.saveUpdateValue",
+            args: ["{that}", "{editPrefs}"]
+        },
+        cancelEditing: {
+            funcName: "gpii.devpmt.prefSettingAdjuster.cancelEditing",
+            args: ["{that}"]
+        },
+        watchInputKeys: {
+            funcName: "gpii.devpmt.prefSettingAdjuster.watchInputKeys",
+            args: ["{that}", "{arguments}.0"]
+        },
+        updateBlankDisabling: {
+            funcName: "gpii.devpmt.prefSettingAdjuster.updateBlankDisabling",
+            args: ["{that}"]
+        }
+    },
+    markupEventBindings: {
+        okButton: {
+            method: "click",
+            args: "{that}.saveUpdateValue"
+        },
+        cancelButton: {
+            method: "click",
+            args: "{that}.cancelEditing"
+        },
+        valueInput: {
+            method: "keypress",
+            args: "{that}.watchInputKeys"
+        }
+    },
+    listeners: {
+        onMarkupRendered: [{
+            funcName: "fluid.focus",
+            args: ["{that}.dom.valueInput"]
+        },
+        {
+            func: "{that}.updateBlankDisabling"
+        },
+        {
+            funcName: "gpii.devpmt.prefSettingAdjuster.updateBooleanValueLabel",
+            args: ["{that}"]
+        }]
+    },
+    modelListeners: {
+        "current.blank": {
+            func: "{that}.updateBlankDisabling"
+        },
+        "current.value": [{
+            funcName: "gpii.devpmt.prefSettingAdjuster.updateBooleanValueLabel",
+            args: ["{that}"]
+        }]
+    }
+});
+
+gpii.devpmt.prefSettingAdjuster.updateBooleanValueLabel = function (that) {
+    // if (that.model.metadata.schema.type === "boolean") {
+    jQuery("#pmt-boolean-value").css("display","inline");
+    jQuery("#pmt-boolean-value").css("color","black");
+    if (that.model.current.value === true)  {
+        jQuery("#pmt-boolean-value").html("true");
+    }
+    else if (that.model.current.value === false) {
+        jQuery("#pmt-boolean-value").html("false");
+    }
+};
+
+/**
+ * Updates the UI based on whether Blank is selected.
+ * If blank is selected, then the rest of the editing
+ * apparatus is disabled.
+ */
+gpii.devpmt.prefSettingAdjuster.updateBlankDisabling = function (that) {
```

Comment:
Rather than passing in `that`, arguments can be more specific with what to be handled such as `function (valueInput, disableState)`. The corresponding invoker arg list can be `[{that}.dom.valueInput, "{that}.model.current.blank"]`.

Reply:
Actually, a more declarative way could be:

```
invokers: {
    updateBlankDisabling: {
        "this": "{that}.dom.valueInput",
        "method": "prop",
        "args": "{that}.model.current.blank"
    }
}
```

Reply:
Thanks! I had forgotten about this syntax for calling jQuery stuff... it's a little odd to look at, but I moved the comment up above, and got rid of the previous function.